tests/helpers/convert_ipynb_to_script.py

```diff
@@ -0,0 +1,346 @@
+        elif cell_type == "code":
+            source: list[str] = cell["source"]
+            if filter_out_lines:
+                source = [
+                    f"#{line}"
+                    if any(
+                        line.startswith(filter_prefix)
+                        for filter_prefix in filter_out_lines_set
+                    )
+                    else line
+                    for line in source
+                ]
+            result.append(f'{header_comment}\n{"".join(source)}')
+
+    if disable_plots:
+        result = disable_plots_in_script(result)
+        result = DISABLE_PLOTS_WARNING + result
+
+    return "\n\n".join(result)
+
+
+def process_file(
+    in_file: str,
+    out_file: str | None = None,
+    strip_md_cells: bool = False,
+    header_comment: str = r"#%%",
+    disable_plots: bool = False,
+    filter_out_lines: str | typing.Sequence[str] = ("%", "!"),
+):
+    print(f"\tProcessing {in_file}...", file=sys.stderr)
+    assert os.path.exists(in_file), f"File {in_file} does not exist."
+    assert os.path.isfile(in_file), f"Path {in_file} is not a file."
+    assert in_file.endswith(".ipynb"), f"File {in_file} is not a Jupyter Notebook."
+
+    with open(in_file, "r") as file:
+        notebook: dict = json.load(file)
+
+    try:
+        converted_script: str = convert_ipynb(
+            notebook=notebook,
+            strip_md_cells=strip_md_cells,
+            header_comment=header_comment,
+            disable_plots=disable_plots,
+            filter_out_lines=filter_out_lines,
+        )
+    except AssertionError as e:
+        print(f"Error converting {in_file}: {e}", file=sys.stderr)
+        raise e
+
+    if out_file:
+        with open(out_file, "w") as file:
+            file.write(converted_script)
+    else:
+        print(converted_script)
+
+
+def process_dir(
+    input_dir: str,
+    output_dir: str,
+    strip_md_cells: bool = False,
+    header_comment: str = r"#%%",
+    disable_plots: bool = False,
+    filter_out_lines: str | typing.Sequence[str] = ("%", "!"),
+):
+    """Convert all Jupyter Notebooks in a directory to scripts.
+
+    # Arguments
+        - `input_dir: str`: Input directory.
+        - `output_dir: str`: Output directory.
+        - `strip_md_cells: bool = False`: Remove markdown cells from the output script.
+        - `header_comment: str = r'#%%'`: Comment string to separate cells in the output script.
+        - `disable_plots: bool = False`: Disable plots in the output script.
+        - `filter_out_lines: str|typing.Sequence[str] = ('%', '!')`: comment out lines starting with these strings (in code blocks).
+            if a string is passed, it will be split by char and each char will be treated as a separate filter.
+    """
+
+    assert os.path.exists(input_dir), f"Directory {input_dir} does not exist."
+    assert os.path.isdir(input_dir), f"Path {input_dir} is not a directory."
+
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
```

Comment:
re: https://github.com/AISC-understanding-search/maze-transformer/pull/153#issuecomment-1500459852 @valedan this line should handle it, I think?

Reply:
[c636566](https://github.com/AISC-understanding-search/maze-transformer/pull/153/commits/c636566a49403b52bf3401fbb17a14f6cc98e8a1) sets `exist_ok=True`, but that won't resolve the issue you're having. It works in CI so I'm really not sure what could be the problem

Reply:
Actually this is probably related to the warning:  
`realpath: illegal option -- -`

at the beginning of the makefile, we define  
`ROOT_RELATIVE_TO_NOTEBOOKS := $(shell realpath --relative-to=$(NOTEBOOKS_DIR) .)`
which is then used to specify output file location in the `test_notebooks` recipe

I assumed that whatever version of `realpath` I have and the CI machine has would be universal, but I guess that's not the case! I think the easiest fix would be to move the majority of `test_notebooks` to a script as per [your suggestion](https://github.com/AISC-understanding-search/maze-transformer/pull/153#discussion_r1160834381), since `ROOT_RELATIVE_TO_NOTEBOOKS` isn't used anywhere else.

Reply:
@valedan , this should be fixed as of [7307f05](https://github.com/AISC-understanding-search/maze-transformer/pull/153/commits/7307f055a34b12256e2738ac7060ad676c365116)

Let me know if this works on your machine!